diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index a9d3b9f..601beb3 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -71,3 +71,5 @@ else()
 endif()
 
 target_compile_options(${EXE_NAME} PRIVATE -O3 -mtune=cortex-a55 -Wall -DDEBUG_LOG)
+
+install(TARGETS ${EXE_NAME} RUNTIME)
diff --git a/how-to/sample_app/src/CMakeLists.txt b/how-to/sample_app/src/CMakeLists.txt
index e97a6a4..713191f 100644
--- a/how-to/sample_app/src/CMakeLists.txt
+++ b/how-to/sample_app/src/CMakeLists.txt
@@ -83,3 +83,4 @@ if(OpenCV_FOUND)
   target_link_libraries(${EXE_NAME} PRIVATE ${OpenCV_LIBS})
 endif()
 
+install(TARGETS ${EXE_NAME} RUNTIME)
diff --git a/how-to/sample_app/src/util/system_analyzer.h b/how-to/sample_app/src/util/system_analyzer.h
index 5b5aaf5..389ec26 100644
--- a/how-to/sample_app/src/util/system_analyzer.h
+++ b/how-to/sample_app/src/util/system_analyzer.h
@@ -37,6 +37,7 @@
 #include <sys/statvfs.h>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 
 class LinuxSystemAnalyzer
